### PR TITLE
dd: Update dd_get_owner to handle error return values

### PR DESCRIPTION
**Pull request by mgrabovsky**
Takeover from #570.

Pretty much just rebased on top of master.

Follow-up: abrt/abrt#1484

### CODE REVIEW

File: src/lib/dump_dir.c [hunk 1]
```diff
@@ -755,10 +758,11 @@ uid_t dd_get_owner(struct dump_dir *dd)
 
     if (ret < 0)
     {
-        if (ret != -ENOENT)
-            return ret;
+        if (errorno && ret != -ENOENT)
+            *errorno = ret;
```
Comment (ernestask) [inline src/lib/dump_dir.c:762]: I’m wondering… What do you think about simply using `errno`? There’s nothing that prevents us from doing so and documenting the magic `(uid_t) -1) value to mean “check errno”. 

Realistically, you would only reach the millions in the container world, but I see that at least my Fedora installation limits the subordinate IDs to 600100000 (which is well below `UINT32_MAX` and `INT32_MAX`).

Additional note is that the signedness of `uid_t` is undefined: https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/sys_types.h.html, which would explain `MAX_UID_T`.

**Comment by mzidek-gh** [comment]
Hi, not sure if this is a concern, but this change breaks the ABI. IIUC this is public symbol for libreport.

**Comment by mgrabovsky** [comment]
> Hi, not sure if this is a concern, but this change breaks the ABI. IIUC this is public symbol for libreport.

Yeah, I don't think we've been any consistent in this regard until now, so… If I recall correctly, nobody else is using these APIs (repoquery'd Fedora repos), so we should be relatively safe.

Of course, it'd be safer to bump the version accordingly whenever a breaking change occurs.

**Comment by mgrabovsky** [comment]
OK, I updated both pull requests to use `errno`. Contrary to my belief, it is already being used in the codebase.

**Comment by ernestask** [comment]
Holy wow, I didn’t expect that.

**Pull request merged**